src/main/java/com/zufar/icedlatte/auth/google/endpoint/AuthEndpoint.java

```diff
@@ -0,0 +1,189 @@
+import com.zufar.icedlatte.security.api.UserRegistrationService;
+import com.zufar.icedlatte.security.dto.UserAuthenticationRequest;
+import com.zufar.icedlatte.security.dto.UserRegistrationRequest;
+import com.zufar.icedlatte.user.entity.UserEntity;
+import com.zufar.icedlatte.user.repository.UserRepository;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import com.google.gson.Gson;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@Validated
+@RequestMapping(value = AuthEndpoint.GOOGLE_AUTH)
+public class AuthEndpoint {
+
+  @Autowired Environment environment;
+
+  @Value("${google.auth.server.url}")
+  public String authorizationServerUrl;
+
+  @Value("${google.client-id}")
+  public String clientId;
+
+  @Value("${google.client-secret}")
+  public String clientSecret;
+
+  @Value("${google.scope}")
+  public String scope;
+
+  @Value("${google.authenticateUri}")
+  public String authenticateUri;
+
+  @Value("${google.redirectUri}")
+  String redirectUri;
+
+  public static final String GOOGLE_AUTH = "/api/v1/3part-auth/google";
+
+  private final UserRepository userCrudRepository;
+  private final UserRegistrationService userRegistrationService;
+
+  @GetMapping()
+  public ResponseEntity<?> googleAuth() {
+
+    log.info("Received the request to add a google auth");
+    String redirectAuthUri =
+        "http://localhost"
+            + ":"
+            + environment.getProperty("server.port")
+            + GOOGLE_AUTH
+            + "/callback";
+    String url =
+        authorizationServerUrl
+            + "?"
+            + "scope="
+            + scope
+            + "&"
+            + "access_type=offline&"
+            + "include_granted_scopes=true&"
+            + "response_type=code&"
+            + "state=state_parameter_passthrough_value&"
+            + "redirect_uri="
+            + redirectAuthUri
+            + "&"
+            + "client_id="
+            + clientId;
+    return ResponseEntity.status(HttpStatus.TEMPORARY_REDIRECT).header("Location", url).build();
+  }
+
+  @GetMapping("/callback")
```

Comment:
I would add wrappers for these methods to `UserSecurityEndpoint` and moved `AuthEndpoint.java` to `src/main/java/com/zufar/icedlatte/security/api/google/GoogleAuth.java`

Also, are these endpoint available in Swagger?

Reply:
In my opinion, this method is not needed in Swagger because it will not be possible to use it due to the redirect to the Google website.